### ISSUE TITLE
Why does this code not work with mode=0?

### ISSUE THREAD

**Issue by yxchng** [fchollet/keras#3369]
The error:
You are attempting to share a '
                                    'same `BatchNormalization` layer across '
                                    'different data flows. '
                                    'This is not possible. '
                                    'You should use `mode=2` in '
                                    '`BatchNormalization`, which has '
                                    'a similar behavior but is shareable '
                                    '(see docs for a description of '
                                    'the behavior).')

<img width="719" alt="screen shot 2016-08-01 at 9 42 04 pm" src="https://cloud.githubusercontent.com/assets/10518587/17295829/d2316fd0-5830-11e6-9a03-d02c5aff309d.png">

Is it because of the way Batchnorm is implemented in Keras or is there something wrong with the code. I read the doc and I think I want to use the average of the mean and std to normalize in test which is mode=0.
